internal/cmd/test.go

```diff
@@ -74,12 +82,19 @@ func CMDServerTestingMain(suite *testing.M) {
 	exit.Exit(code)
 }
 
-func RunCMDTest(test *testing.T, mainFunc func(), args []string, logLevel log.LogLevel) (string, string, int, error) {
+func RunCMDTest(test *testing.T, mainFunc func(), args []string, logLevel log.LogLevel, removeTempDirsAfterEachTest bool) (string, string, int, error) {
 	// Suppress exits to capture exit codes.
 	exit.SetShouldExitForTesting(false)
 	defer exit.SetShouldExitForTesting(true)
 
+	// Retain temp dirs to use after tests complete.
 	tempDir := util.MustMkDirTemp("autograder-testing-cmd-")
+	util.SetShouldRemoveTempDirs(false)
+	if removeTempDirsAfterEachTest {
+		defer util.RemoveRecordedTempDirs()
+	}
+	defer util.SetShouldRemoveTempDirs(true)
```

Comment:
This is a very special situation that only one test needs.
We can just set the value in that one test.